__tests__/app.test.js

```diff
@@ -34,111 +34,167 @@ describe("GET /api/topics", () => {
+            });
+         });
+   });
+   test("404: responds with a error message when article_id doesn't exist", () => {
+      const ARTICLE_ID = 88;
+      return request(app)
+         .get(`/api/articles/${ARTICLE_ID}`)
+         .expect(404)
+         .then((response) => {
+            expect(response.body.msg).toBe(
+               "Article not found! Please try again"
+            );
+         });
+   });
+   test("400: responds with a error message when client pass article_id which is not a number", () => {
+      const ARTICLE_ID = true;
+      return request(app)
+         .get(`/api/articles/${ARTICLE_ID}`)
+         .expect(400)
+         .then((response) => {
+            expect(response.body.msg).toBe(
+               "Please type/select a number to choose an article"
+            );
+         });
+   });
+});
+describe("8. GET /api/articles", () => {
+   test("200: responds with an array of articles sorted by date in descending order", () => {
+      return request(app)
+         .get("/api/articles")
+         .expect(200)
+         .then((response) => {
+            expect(typeof response.body).toBe('object')
+            expect(Array.isArray(response.body)).toBe(true);
+            expect(response.body.length > 0).toBe(true);
+            expect(response.body).toBeSortedBy('created_at', {descending: true,})
+            console.log(response.body, '<<< my response.body')
+            response.body.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    comment_count: expect.any(Number),
+                 });
             })
-        })
-    })
-    test('404: responds with a error message when article_id doesn\'t exist', () => {
-        const ARTICLE_ID = 88;
-        return request(app)
-        .get(`/api/articles/${ARTICLE_ID}`)
-        .expect(404)
-        .then((response) => {
-            expect(response.body.msg).toBe('Article not found! Please try again')
-        })
-    })
-    test('400: responds with a error message when client pass article_id which is not a number', () => {
-        const ARTICLE_ID = true;
-        return request(app)
-        .get(`/api/articles/${ARTICLE_ID}`)
-        .expect(400)
+        });
+   });
+   test("200: filter objects by topic given", () => {
+    return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
         .then((response) => {
-            expect(response.body.msg).toBe('Please type/select a number to choose an article')
+            expect(typeof response.body).toBe('object')
+            expect(Array.isArray(response.body)).toBe(true);
+            expect(response.body.length > 0).toBe(true);
```

Comment:
I would make this expect more specific, for example how many articles should come back with the topic of cats? This expect statement doesn't tell me anything other than it's greater than 0, however for a more confident test let's see a number here which is representative of how many actual articles are returned... e.g. if there are 3 articles with the topic of 'cats' in the database, then this expect statement should be `expect(response.body.length).toBe(3)`; or you could use another jest matcher `expect(response.body).toHaveLength(3)`

Reply:
It makes total sense. I will study more about jest matchers this weekend!
Done